### PR TITLE
Whatsnew reorganization

### PR DESCRIPTION
**Pull request by TomAugspurger**
Changes

1. Adds a highlights section (open to suggestions on what should be added / removed from here)
2. Reorders the enhcancements section to be
    1. Highlighted features
    2. Other features (sections)
    3. List of other new features (one-liners)
3. Reorders the "Backwards incompatible changes" section. We had two lists of "other incompatible changes" that I've merged into one at the end of the section.
4. Moved extension type changes out of the "Backwards incompatible changes section". 

cc @jorisvandenbossche

### CODE REVIEW

**Review by jorisvandenbossche — APPROVED** [review]
Thanks! That more or less exactly the clean-up I also wanted to do :)

File: doc/source/whatsnew/v0.24.0.rst [hunk 1]
```diff
@@ -10,92 +10,16 @@ What's New in 0.24.0 (January XX, 2019)
...
-   on the underlying elements (scalars) of the ``ExtensionArray``.
-
-See the :ref:`ExtensionArray Operator Support
-<extending.extension.operator>` documentation section for details on both
-ways of adding operator support.
+* :ref:`Optional Nullable Integer Support <whatsnew_0240.enhancements.intna>`
+* :ref:`New APIs for accessing the array backing a Series or Index <whatsnew_0240.values_api>`
+* :ref:`A new top-level method for creating arrays <whatsnew_0240.enhancements.array>`
+* :ref:`Store Interval and Period data in a Series or DataFrame <whatsnew_0240.enhancements.interval>`
+* :ref:`Support for joining on two MultiIndexes <whatsnew_0240.enhancements.join_with_two_multiindexes>`
```
[the remark this answers is not in the document]
Comment (TomAugspurger) [reply doc/source/whatsnew/v0.24.0.rst]: Highlight: If you were using SparseArrays we probably broke your code :)

I'm planning to go into much more detail about the extension array stuff in the blog post, so slightly inclined to not say too much here, as it's probably not directly relevant for the average user.

**Comment by jreback** [comment]
lgtm. after CI fixes are merge, let's repush and merge on green.

**Comment by TomAugspurger** [comment]
Gonna just fix the merge conflict and merge if that's OK.

File: doc/source/whatsnew/v0.24.0.rst [hunk 2]
```diff
@@ -1233,6 +1160,58 @@ Other API Changes
...
 
+
+.. _whatsnew_0240.api.extension:
+
+ExtensionType Changes
+^^^^^^^^^^^^^^^^^^^^^
+
+**Equality and Hashability**
+
+Pandas now requires that extension dtypes be hashable. The base class implements
```
Comment (h-vetinari) [inline doc/source/whatsnew/v0.24.0.rst]: Seems I'm a bit too late to mention this on time. In https://github.com/pandas-dev/pandas/issues/22476#issuecomment-440349264, we talked about this line.

How about: Pandas now requires that extension dtypes be hashable *(i.e. the respective `ExtensionDtype`-objects; hashability is not a requirement for the values of the corresponding `ExtensionArray`)*.

**Comment by h-vetinari** [comment]
@TomAugspurger 

Seems that comment about the hashable dtype we talked about hasn't made it into master yet. Just mentioning if you still wanna do that before release.

**Comment by TomAugspurger** [comment]
Hmm keep forgetting to update it.

**Pull request merged**